src/utils/optionsdict.cc

```diff
@@ -322,4 +322,14 @@ void OptionsDict::AddSubdictFromString(const std::string& str) {
   parser.ParseMain(this);
 }
 
+std::string OptionsDict::FindNotUsed() const {
```

Comment:
I'd make a function like `CheckAllOptionsRead()`, which throws exception itself, but probably matter of taste.

Reply:
...And would make this function get `(std::string& path_from_parent)` (for `--backend-opts=a(cpu=4)` it would be "a", so that it could complain about `a.cpu` being unknown).

Reply:
...So it would be nice to also check subdicts, and it would also work if value is accessed through some child..
Cases like that will fail though:
`gpu=1,(gpu=2),(gpu=3)`, if gpu=1 is not used. But probably that's good.

Reply:
Hmm, I didn't consider the subdict case - need more testing. The reason for not throwing the exception directly was to have a backend-opts specific error message.

Reply:
Maybe throw another exception (but still derived from `lczero::Exception`) and rethrow converted to more readable exception in the calling code?
Maybe it's overengineering though.